google-cloud-spanner/src/main/java/com/google/cloud/spanner/Mutation.java

```diff
@@ -386,8 +388,10 @@ private boolean areValuesEqual(List<Value> values, List<Value> otherValues) {
 
   private boolean isNaN(Value value) {
```

Comment:
It's getting difficult to read this method now with so many nested operators. Can we break the method into smaller methods (1 per type).

Reply:
Done